internal/resources/grafana/resource_data_source_permission.go

```diff
@@ -82,25 +86,24 @@ Manages the entire set of permissions for a datasource. Permissions that aren't
 }
 
 func UpdateDatasourcePermissions(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client, orgID := DeprecatedClientFromNewOrgResource(meta, d)
+	client, orgID := OAPIClientFromNewOrgResource(meta, d)
 
 	var list []interface{}
 	if v, ok := d.GetOk("permissions"); ok {
 		list = v.(*schema.Set).List()
 	}
 
-	_, datasourceIDStr := SplitOrgResourceID(d.Get("datasource_id").(string))
-	datasourceID, _ := strconv.ParseInt(datasourceIDStr, 10, 64)
-
-	dataSource, err := client.DataSource(datasourceID)
+	_, datasourceID := SplitOrgResourceID(d.Get("datasource_id").(string))
+	resp, err := client.Datasources.GetDataSourceByID(datasourceID)
```

Comment:
We should probably extend the data source resource to include UID, so that we can operate with UIDs directly.

Reply:
Indeed. We can definitely support both in the read function: https://github.com/grafana/terraform-provider-grafana/blob/master/internal/resources/grafana/resource_folder.go#L122